templates/nextflow-s3-template.yaml

```diff
@@ -0,0 +1,139 @@
+AWSTemplateFormatVersion: 2010-09-09
+# Description: Custom Nextflow Bucket
+
+# Parameters
+Parameters:
+  Department:
+    # Description: 'The department for this resource'
+    Type: String
+    AllowedPattern: '^\S*$'
+    ConstraintDescription: 'Must be string with no spaces'
+  Project:
+    # Description: 'The name of the project that this resource is used for'
+    Type: String
+    AllowedPattern: '^\S*$'
+    ConstraintDescription: 'Must be string with no spaces'
+  OwnerEmail:
+    # Description: 'Email address of the owner of this resource'
+    Type: String
+    AllowedPattern: '^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$'
+    ConstraintDescription: 'Must be an acceptable email address syntax(i.e. joe.smith@sagebase.org)'
+  BucketName:
+    Type: String
+    # Description: (Optional) Name of the created bucket.
+    # Default: ""
+  BucketAccess:
+    Type: CommaDelimitedList
+    # Default: "[]"
```

Comment:
I think it would be a good idea to use role-based access for this bucket. What I mean is that the current access is divided up by what a user can do -- can they list buckets? Can they read objects in a bucket? Can they write to a bucket? Instead I suggest that you think of it as three levels of role: 
1. Admin-level access has `s3:*` for this bucket and its objects -- this should be the CI user, root, and admin roles. I can help you set these. I think for this level you can set the principals directly rather than parameterize it.
2. Read-write access: they can list buckets, list objects, read objects, and write to the bucket. This would be an optional parameter, so it should have a Default value of empty (i.e. not required).
3. Read-only access: they can list and read. This would also be an optional parameter.

Let's discuss this more in person.